docs/fsharp-cheatsheet.md

```diff
@@ -540,17 +537,106 @@ Another way of implementing interfaces is to use *object expressions*.
 
 *Partial active patterns*:
 
-    let (|DivisibleBy|_|) by n = 
+    let (|DivisibleBy|_|) by n =
         if n % by = 0 then Some DivisibleBy else None
-    
-    let fizzBuzz = function 
-        | DivisibleBy 3 & DivisibleBy 5 -> "FizzBuzz" 
-        | DivisibleBy 3 -> "Fizz" 
-        | DivisibleBy 5 -> "Buzz" 
+
+    let fizzBuzz = function
+        | DivisibleBy 3 & DivisibleBy 5 -> "FizzBuzz"
+        | DivisibleBy 3 -> "Fizz"
+        | DivisibleBy 5 -> "Buzz"
         | i -> string i
 
 *Partial active patterns* share the syntax of parameterized patterns but their active recognizers accept only one argument.
 
+<a name="CodeOrganization"></a>Code Organization
+---------
+
+### Modules
+Modules are key building blocks for grouping related concepts; they can contain `types`, `let` bindings, and even other `modules`.
+Modules can be referenced using dot notation or exposed with the `open` keyword. Illustrative-only example:
+
+    module Game =
+        let mutable basePoints = 1
+        type Player = { id: int; score: int }
+        let playerScored player = { player with score = player.score + basePoints }
+
+    let player: Game.Player = { id = 1; score = 0 }
+    let player' = Game.playerScored player  // score = 1
+
+    open Game
+    basePoints <- 2
+    let player'' = playerScored player'  // player''.score = 3
+
+If you have only one module in your file, the `module` can be specified at the top of the file.
+
+    module Functions  // notice there is no '=' when at the top of a file
+
+    let addToFive num = 5 + num
+    let subtractFive num = num - 5
+
+### Namespaces
+Namespaces are simply dotted strings that prefix names. They are placed at the top of the file.
+An inner namespace can be specified in the same file.
+
+    namespace MyNamespace
+
+    namespace MyNamespace.InnerSpace
+
+They can also be part of a module name.
+
+    module MyNamespace.InnerSpace.MyModule
+
+### Open and AutoOpen
+
+The `open` keyword can be used with `module`, `namespace`, and `type`.
+
+    open System.Diagnostics  // open namespace
+    let stopwatch = Stopwatch.StartNew()
+    ---
+    module MyModule =
+        type DU1 = A | B | C
+        type DU2 = D | E | F
+
+    open type MyModule.DU1
+    let du1 = A
+    let duNotDefined = D  // 'D' not defined
+    open MyModule
+    let du2 = D
+    ---
+    [<AutoOpen>]
```

Comment:
need to convey that this is a global thing, that anyone importing the contaiing namespace will have it implicitly opened (and that you might often do an `open` X immediately after declaring it instead to avoid polluting the golbal namespace with implicit things)

Reply:
Coming update.